compiler/ilgen/IlBuilder.cpp

```diff
@@ -1443,10 +1443,50 @@ IlBuilder::Xor(TR::IlValue *left, TR::IlValue *right)
    return returnValue;
    }
 
+TR::Node*
+IlBuilder::shiftOpNodeFromNodes(TR::ILOpCodes op,
+                                TR::Node *leftNode,
+                                TR::Node *rightNode) 
+   {
+   TR::DataType leftType = leftNode->getDataType();
+   TR::DataType rightType = rightNode->getDataType();
+   TR_ASSERT(leftType.isIntegral() && rightType.isInt32(), "shift operation first operand must be an integer, and shift amount must be 32-bit integer");
+
+   return TR::Node::create(op, 2, leftNode, rightNode);
+   }
+
+TR::IlValue *
+IlBuilder::shiftOpFromNodes(TR::ILOpCodes op,
+                            TR::Node *leftNode,
+                            TR::Node *rightNode) 
+   {
+   TR::Node *result = shiftOpNodeFromNodes(op, leftNode, rightNode);
+   TR::IlValue *returnValue = newValue(result->getDataType(), result);
+   return returnValue;
+   } 
+
+TR::IlValue *
+IlBuilder::shiftOpFromOpMap(OpCodeMapper mapOp,
+                            TR::IlValue *left,
+                            TR::IlValue *right)
+   {
+   TR::Node *leftNode = loadValue(left);
+   TR::DataType leftType = leftNode->getDataType();
+   TR_ASSERT(leftType.isIntegral(), "left operand of shift must be integer type");
+
+   TR::Node *rightNode = loadValue(right);
+   if (!rightNode->getDataType().isInt32())
```

Comment:
@charliegracie where do you want to shift that value, Charlie? :)  That would be 9,223,372,036,854,775,807 bit positions!

Note that it will convert to 32-bits for you, so will 2B shifts be enough :) ?